### DISCUSSION TITLE
help-“The rule_id is lost after a restart”

### DISCUSSION THREAD

**Discussion by zhou0bin** [lf-edge/ekuiper#3140] [Q&A]
I created a rule, and after a restart, the rule's effect is still present, but the rule_id is lost. For example, executing a drop command via the CLI gives a 'not found' error, but trying to recreate the rule shows 'store the rule error: Item rule2 already exists.' Although it shows an error, the rule works normally after being recreated.

log：time="2024-08-23T17:26:23+08:00" level=info msg="provision sink log_0 with props map[]" file="planner/planner_sink.go:91" rule=
time="2024-08-23T17:26:23+08:00" level=info msg="create bytes sink node log_0" file="node/sink_node.go:172" rule=
time="2024-08-23T17:26:23+08:00" level=info msg="create sink node log_0 with isRetry false, resendInterval 0, bufferLength 1024" file="node/sink_node.go:61" rule=
time="2024-08-23T17:26:23+08:00" level=info msg="provision sink mqtt_1 with props map[server:tcp://broker.emqx.io:1883 topic:demoSink]" file="planner/planner_sink.go:91" rule=
time="2024-08-23T17:26:23+08:00" level=info msg="create bytes sink node mqtt_1" file="node/sink_node.go:172" rule=
time="2024-08-23T17:26:23+08:00" level=info msg="create sink node mqtt_1 with isRetry false, resendInterval 0, bufferLength 1024" file="node/sink_node.go:61" rule=

The version is the latest master.

**Comment by ngjaying** [comment]
The rule json does not specify an id. Try to make sure rule json has defined the id. We also add validation to avoid invalid rule json string being stored.

**Reply by zhou0bin** [reply]
It's done; thank you.
